mantle/kola/harness.go

```diff
@@ -342,26 +349,54 @@ func parseDenyListYaml(pltfrm string) error {
 
 	plog.Debug("Parsed kola-denylist.yaml")
 
-	// Get stream and arch
-	pathToManifest := filepath.Join(Options.CosaWorkdir, "src/config/manifest.yaml")
-	manifestFile, err := ioutil.ReadFile(pathToManifest)
+	// Look for the right manifest, taking into account the variant
+	var manifest ManifestData
+	var pathToManifest string
+	pathToInitConfig := filepath.Join(Options.CosaWorkdir, "src/config.json")
+	initConfigFile, err := ioutil.ReadFile(pathToInitConfig)
 	if os.IsNotExist(err) {
-		return nil
+		// No variant config found. Let's read the default manifest
+		pathToManifest = filepath.Join(Options.CosaWorkdir, "src/config/manifest.yaml")
 	} else if err != nil {
+		// Unexpected error
+		return err
+	} else {
+		// Figure out the variant and read the corresponding manifests
+		var initConfig InitConfigData
+		err = json.Unmarshal(initConfigFile, &initConfig)
+		if err != nil {
+			return err
+		}
+		pathToManifest = filepath.Join(Options.CosaWorkdir, fmt.Sprintf("src/config/manifest-%s.yaml", initConfig.ConfigVariant))
+	}
+	manifestFile, err := ioutil.ReadFile(pathToManifest)
+	if err != nil {
 		return err
 	}
-
-	var manifest ManifestData
 	err = yaml.Unmarshal(manifestFile, &manifest)
 	if err != nil {
 		return err
 	}
 
+	// Get the stream, distro & version variables from the manifest
 	stream := manifest.Variables.Stream
+	distro := manifest.Variables.Distro
+	version := manifest.Variables.Version
+
+	// Generate the variant names that we use in the denylist
```

Comment:
I'm confused by this. Didn't we already parse the variant higher up?

Reply:
For kola, we need to resolve the os version, otherwise we will have to include all variants in the denylist which is not great.

Reply:
Makes sense. My only concern is that we're calling "variant" a different set of possible names in the denylist vs in `cosa init`, which doesn't seem great.

How about instead we just have the keyname used in the denylist be `distro-versions`? And it's always `${distro}-${version}`, even in the CentOS case.

Reply:
It's tempting to make this generic over any variable defined in the manifest, but that feels like overkill.

Reply:
In this case we would have `centos-9`, which we have nowhere else. I agree it's not great in both cases.

Reply:
The other option is to get rid of those variables and add a new version one that has both the distro & version in its name

Reply:
I'll do that